tests/labs/lie/functional/common.py

```diff
@@ -126,3 +126,60 @@ def func(matrix: torch.Tensor):
         return module._left_project_autograd_fn(group, matrix[sels, ..., sels, :, :])
 
     return func
+
+
+def check_jacrev_unary(group_fns, dim, batch_size, name):
+    if not hasattr(torch, "vmap"):
+        return
+
+    test_fn = getattr(group_fns, name)
+
+    fn_input = (
+        torch.randn(batch_size, dim) if name == "exp" else group_fns.rand(batch_size)
+    )
+
+    def f(t):
+        return group_fns.log(test_fn(t.unsqueeze(0))).squeeze(0)
+
+    j = torch.vmap(torch.func.jacrev(f))(fn_input)
+    jac_vmap = group_fns.left_project(fn_input, j) if name != "exp" else j
+
+    jlog, jtest = [], []
+    group_fns.log(test_fn(fn_input, jacobians=jtest), jacobians=jlog)
+    jac_analytic = jlog[0] @ jtest[0]
+
+    torch.testing.assert_close(jac_vmap, jac_analytic)
+
+
+def check_jacrev_binary(group_fns, batch_size, name):
+    if not hasattr(torch, "vmap"):
+        return
+
+    test_fn = getattr(group_fns, name)
+
+    fn_inputs = (
+        (group_fns.rand(batch_size), torch.randn(batch_size, 3))
+        if name == "transform_from"
+        else (group_fns.rand(batch_size), group_fns.rand(batch_size))
+    )
+
+    def f(t1, t2):
+        op_out = test_fn(t1.unsqueeze(0), t2.unsqueeze(0))
+        if name == "compose":
+            op_out = group_fns.log(op_out)
+        return op_out.squeeze(0)
+
+    jacs_vmap = []
+    for i in range(2):
+        j = torch.vmap(torch.func.jacrev(f, i))(fn_inputs[0], fn_inputs[1])
+        if fn_inputs[i].ndim == 3:  # group input
+            j = group_fns.left_project(fn_inputs[i], j)
```

Comment:
We don't need `left_project`  for the 2nd argument of `transform_from`.

Reply:
The second argument of `transform_from` is `ndim == 2` , so `left_project` is not called in that case.